WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift

```diff
@@ -561,6 +565,31 @@ private extension CardReaderConnectionController {
         }
     }
 
+    private func showConnectionFailed(error: Error) {
+        guard let from = fromController else {
+            return
+        }
+
+        let continueSearch = {
+            self.state = .retry
+        }
+
+        let cancelSearch = {
+            self.state = .cancel
+        }
+
+        guard case CardReaderServiceError.connection(let underlyingError) = error else {
+            return alerts.connectingFailed(from: from, continueSearch: continueSearch, cancelSearch: cancelSearch)
+        }
+
+        switch underlyingError {
+        case .incompleteStoreAddress(adminUrl: _):
+            alerts.connectingFailedMissingAddress(from: from, continueSearch: continueSearch, cancelSearch: cancelSearch)
```

Comment:
In light of #5560, which I noticed after I reviewed this PR, I'd opt for "Incomplete" instead of "Missing"

Reply:
Because there's a string of PRs to merge here, to avoid messy merges I'm going to open issues for minor changes and address them next, after merging all PRs. If there's anything fundamental, I won't merge until fixing, but from a skim it doesn't look like there is. See #5567